source/robot_model/include/robot_model/Model.hpp

```diff
@@ -264,6 +264,13 @@ class Model {
    */
   static bool create_urdf_from_string(const std::string& urdf_string, const std::string& desired_path);
 
+  /**
+   * @brief Compute the minimum distance between the robot links
+   * @param joint_positions state_representation object containing the joint positions of the robot
+   * @return the matrix containing the minimum distance between the robot links
+   */
+  Eigen::MatrixXd compute_minimum_distance(const state_representation::JointPositions& joint_positions); 
```

Comment:
nitpick: I think the function name should at least be `distances` not `distances`, but preferably also includes "collision" somewhere. Let me know if you think it's too verbose like this:
```suggestion
  Eigen::MatrixXd compute_minimum_collision_distances(const state_representation::JointPositions& joint_positions); 
```